src/engine/SpatialJoinAlgorithms.cpp

```diff
@@ -37,25 +38,99 @@ SpatialJoinAlgorithms::SpatialJoinAlgorithms(
       geometries_{qec->getAllocator()} {}
 
 // ____________________________________________________________________________
-util::geo::I32Box SpatialJoinAlgorithms::libspatialjoinParse(
-    bool leftOrRightSide, const IdTable* idTable, ColumnIndex column,
-    sj::Sweeper& sweeper, size_t numThreads) const {
+std::optional<util::geo::DBox> SpatialJoinAlgorithms::prefilterBoxToLatLng(
+    std::optional<util::geo::I32Box> prefilterBox) {
+  if (!prefilterBox.has_value()) {
+    return std::nullopt;
+  }
+
+  // Reverse projection applied by `sj::WKTParser`: convert coordinates from web
+  // mercator int32 to normal lat-long double coordinates.
+  auto projection = [](const util::geo::I32Point& p) {
+    return util::geo::webMercToLatLng<double>(
+        static_cast<double>(p.getX()) / PREC,
+        static_cast<double>(p.getY()) / PREC);
+  };
```

Comment:
where is the `PREC` constant defined? Is that an internal of the `libspatialjoin?

+ This function looks like it should be a function of the libspatialjoin, as it converts a libspatialjoin type to another, at least factor it out into a separate helper.

Reply:
helper header I meant, with tests that this is the inverse of what libspatialjoin does.